torch/fx/experimental/proxy_tensor.py

```diff
@@ -55,6 +55,15 @@ def decompose(decomposition_table):
 proxy_slot = object()
 no_default = object()
 
+def is_sym_node(node):
+    # TODO(whc) what cases does node.meta not have a val?
+    # will all symints have a node.meta['val']? what about all tensors?
```

Comment:
Our tracing invariant is that ALL nodes produced by post-functionalization tracing have a val attached to them. Pre-functionalization tracing we cannot guarantee this for tensors as there is no way to preserve both object identity and metadata mutation in the case you have something like `x = y.resize_(blah)` in the graph (x and y are the same object, but their metadata changes depending on if you're before or after the line. Because we only run this in the partitioner post functionalization there is no problem. However, if you are afraid of this case, we can also introduce an alternate field for meta which just says what the kind of node it is (Tensor or SymInt/SymFloat)

Reply:
the case where this breaks down is tuples.  example:
```
node.op ->"call_function"
node.target -> <OpOverload(op='aten.native_batch_norm', overload='default')>
"val" in node.meta: False
node.users -> {getitem: None, getitem_1: None, getitem_2: None}
```

i wonder if we can tag this case during set_meta, using anything other than "all users are getitem' heuristic.  Or maybe that heuristic is good enough?

Reply:
We would just do it before we trace the `getitem` unpacking: https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/proxy_tensor.py#L161

so
```
    if isinstance(inner_res, tuple) or isinstance(inner_res, list):
        (+) proxy_res.node.meta['val'] = inner_res (or maybe something else, dunno)
```